resources/models.py

```diff
@@ -451,6 +453,45 @@ class Meta:
         abstract = True
 
 
+class BerthManager(models.Manager):
+    def get_queryset(self):
+        """
+        The QuerySet annotates whether a berth is available or not. For this,
+        it considers the following criteria:
+            - If there are leases associated to the berth
+            - If any lease ends during the current or the last season (previous year)
+                + If a lease ends during the current season:
+                    * It needs to have a "valid" status (DRAFTED, OFFERED, PAID)
+                + If a lease ended during the last season:
+                    * It needs to have a "valid" status (PAID)
+                    * It needs to have renew automatically set
+        """
+        from leases.models import BerthLease
+
+        current_season_end = calculate_berth_lease_end_date()
+
+        in_current_season = Q(end_date=current_season_end)
```

Comment:
should we maybe check for the current season using the start_date? what if the lease starts on the season opening day but is only for 2 months and not till the end of the season? probably:
1. today is before season opening day -> just use calculate start date
2. today is after season start date -> start date after today

Reply:
as discussed on slack: this is not actually the logic we want. no changes needed here.